apiserver/observer/metricobserver/metricobserver.go

```diff
@@ -60,38 +72,14 @@ func NewObserverFactory(config Config) (observer.ObserverFactory, error) {
 		return nil, errors.Annotate(err, "validating config")
 	}
 
-	apiRequestsTotal := prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "juju",
-		Subsystem: "api",
-		Name:      "requests_total",
-		Help:      "Number of Juju API requests served.",
-	}, metricLabelNames)
-
-	apiRequestDuration := prometheus.NewSummaryVec(prometheus.SummaryOpts{
-		Namespace: "juju",
-		Subsystem: "api",
-		Name:      "request_duration_seconds",
-		Help:      "Latency of Juju API requests in seconds.",
-	}, metricLabelNames)
-
-	config.PrometheusRegisterer.Unregister(apiRequestsTotal)
-	if err := config.PrometheusRegisterer.Register(apiRequestsTotal); err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	config.PrometheusRegisterer.Unregister(apiRequestDuration)
-	if err := config.PrometheusRegisterer.Register(apiRequestDuration); err != nil {
-		return nil, errors.Trace(err)
-	}
-
 	// Observer is currently stateless, so we return the same one for each
 	// API connection. Individual RPC requests still get their own RPC
 	// observers.
 	o := &Observer{
 		clock: config.Clock,
 		metrics: metrics{
-			apiRequestDuration: apiRequestDuration,
-			apiRequestsTotal:   apiRequestsTotal,
+			apiRequestDuration: config.MetricsCollector.APIRequestDuration(),
+			apiRequestsTotal:   config.MetricsCollector.APIRequestsTotal(),
```

Comment:
This will be a little messier as we add apiserverRequestDuration.

Reply:
Yeah, I'm going to think about this one.